demuxer.go

```diff
@@ -119,14 +119,16 @@ func (dmx *Demuxer) NextData() (d *DemuxerData, err error) {
 					}
 
 					// Parse data
-					if ds, err = parseData(ps, dmx.optPacketsParser, dmx.programMap); err != nil {
+					var err2 error
+					if ds, err2 = parseData(ps, dmx.optPacketsParser, dmx.programMap); err2 != nil {
 						// We need to silence this error as there may be some incomplete data here
 						// We still want to try to parse all packets, in case final data is complete
 						continue
 					}
 
 					// Update data
 					if d = dmx.updateData(ds); d != nil {
+						err = nil
```

Comment:
Why are you resetting `err` here? 🤔 

imho we still want to return an `ErrNoMorePackets` here so that the developer knows the end of stream has been reached.

Reply:
i'm not sure about this, in my opinion a read function should return either `(data, nil)` or `(nil, error)` - in this case, even if we reset the error, it reappears at the next function call and makes the function return `(nil, error)`

Reply:
I agree with you 👍